library/Fission/DNS.hs

```diff
@@ -0,0 +1,34 @@
+module Fission.DNS (splitRecord) where
+
+import           Fission.Prelude
+
+import           RIO.NonEmpty
+import qualified RIO.Text as Text
+import qualified RIO.List as List
+
+
+-- if less than 256 bytes, return normal. 
+-- if larger then split into chunks & prefix with a 3-char decimal prefix  & semicolon delimiter ("001;")
+splitRecord :: Text -> NonEmpty Text
+splitRecord txt = 
+  if Text.length txt <= 256
+    then 
+      pure txt
+
+    else 
+      Text.chunksOf 252 txt
+      |> List.zip [0..]
+      |> (joinRecord <$>)
+      |> nonEmpty
+      |> maybe (pure "") identity
+
+joinRecord :: (Integer, Text) -> Text
+joinRecord (index, txt) =  
+  textShow index
+  |> padToLen 3 '0'
+  |> (<> ";" <> txt)
+
+padToLen :: Int -> Char -> Text -> Text
+padToLen len char txt = do
+  let paddingLen = max 0 $ len - Text.length txt
+  Text.replicate paddingLen (Text.singleton char) <> txt
```

Comment:
I didn’t even know that Text was a monad :P This version is fine, but you have two other options:

```haskell

padToLen len char txt =
  let paddingLen = max 0 $ len - Text.length txt
  in Text.replicate paddingLen (Text.singleton char) <> txt

padToLen len char txt =
  Text.replicate paddingLen (Text.singleton char) <> txt
  where
    paddingLen = max 0 (len - Text.length txt)
```